### ISSUE TITLE
Pytest 8.1.1  fixture injection compability

### ISSUE THREAD

**Issue by mp2218** [pytest-dev/pytest-bdd#677]
Test which had fixture injection worked fine last week, but now with new pytest release  breaks with following trace:

> def inject_fixture(request: FixtureRequest, arg: str, value: Any) -> None:
>        """Inject fixture into pytest fixture request.
        :param request: pytest fixture request
        :param arg: argument name
        :param value: argument value
        """
       fd = FixtureDef(
            fixturemanager=request._fixturemanager,
            baseid=None,
            argname=arg,
            func=lambda: value,
            scope="function",
            params=None,
        )
E       TypeError: FixtureDef.__init__() got an unexpected keyword argument 'fixturemanager'
/opt/hostedtoolcache/Python/3.11.8/x[64](https://github.com...3#step:13:65)/lib/python3.11/site-packages/pytest_bdd/steps.py:213: TypeError

snippet of fixture injection:

@given("User does not have credit", target_fixture="fielding_page")
def no_money(context_logged_in, base_url):
    return Fielding_Page(context_logged_in.new_page(), base_url)

Version: bdd-7.1.1
`

**Comment by simbs** [comment]
I ran into this yesterday as well. Here's the [relevant pytest change](https://github.com/pytest-dev/pytest/commit/372c17e22899e8e676e105dae714e897426bef86), it's there from version 8.1.1 

I was able to reproduce this locally running the `pytest-bdd` tests with the latest pytest version. I could also fix them by updating the call to `FixtureDef` in `inject_fixture` here: https://github.com/pytest-dev/pytest-bdd/blob/master/src/pytest_bdd/steps.py#L213

but obviously it then breaks compatibility with previous versions of pytest. The `compat.py` file is already addressing what looks like a related compatibility issue, with two different functions depending on whether pytest is below or above 8.1, so creating a function there that instantiates `FixtureDef` differently based on the version worked for me locally.

It's not the nicest solution, but I'm happy to open a PR with these changes if it'll help.
